src/multiplex/client.rs

```diff
@@ -22,10 +24,10 @@ pub trait TagStore<Request, Response> {
     type Tag: Eq;
 
     /// Assign a fresh tag to the given `Request`, and return that tag.
-    fn assign_tag(&mut self, r: &mut Request) -> Self::Tag;
+    fn assign_tag(self: Pin<&mut Self>, r: &mut Request) -> Self::Tag;
```

Comment:
why does this need to be pin?

Reply:
Because we give out `&mut` to it from behind a `Pin`. The alternative would be to require `TagStore + Unpin` everywhere.

Reply:
Specifically, the issue is that the same `T` that is `:TagStore` is _also_ our `Stream + Sink`, so we are going to be giving out `Pin<&mut T>`, which means we can't then hand out `&mut T`.

Reply:
Elaborated on this in the docs for `TagStore` now.